src/python_testing/TC_IDM_4_2.py

```diff
@@ -437,88 +505,88 @@ async def test_TC_IDM_4_2(self):
         except Exception:
```

Comment:
above - I'm pretty sure this is used incorrectly. In my reading, that should be wrapping the function call itself.

except Exception is liable to cause a lint error. It's probably easier to just let that exception do its thing - it will cause the test to fail.

But also...I'm pretty sure that ReadAttribute does not throw a ChipStackError on a read failure. You should get a value decode failure. See read_single_attribute_expect_error function in the matter_testing_support file for the implementation of this.

Reply:
**ValueDecodeFailure** states this:
https://github.com/project-chip/connectedhomeip/blob/6bd80b080eff659ceb3b77e5eb08992b69be505e/src/controller/python/chip/clusters/Attribute.py#L276-L280

It doesn't appear to be an actual exception type.

The error in step 10 is expected to come from invalid report intervals, not invalid clusters/attributes.

Maybe that's why **ValueDecodeFailure** isn't being summoned, **InteractionModelError** isn't being thrown either, even though it's what ReadAttribute claims it throws in the function description.

https://github.com/project-chip/connectedhomeip/blob/6bd80b080eff659ceb3b77e5eb08992b69be505e/src/controller/python/chip/ChipDeviceCtrl.py#L1418-L1419

Both **InteractionModelError** and **ChipStackError** inherit from **ChipStackException** 

https://github.com/project-chip/connectedhomeip/blob/6bd80b080eff659ceb3b77e5eb08992b69be505e/src/controller/python/chip/interaction_model/__init__.py#L92

https://github.com/project-chip/connectedhomeip/blob/6bd80b080eff659ceb3b77e5eb08992b69be505e/src/controller/python/chip/exceptions/__init__.py#L34

So for now I wrapped the ReadAttribute call in a **ChipStackError** exception assert, which is the exception it's actually throwing.